R/make-array.R

```diff
@@ -90,6 +91,16 @@ deriveArray <- function (subvariables, name, ...) {
             .Names=subvarids)),
         list(value=I(subvarids))))
 
+    if (!missing(selections)) {
+        # if there are selections, wrap the array function inside of a
+        # select_categories function
+        if(!is.list(selections)){
+            # if selections is not already a list convert it to one.
+            selections <- as.list(selections)
+        }
+        derivation <- zfunc("select_categories", derivation, list(value=selections))
```

Comment:
I don't think the as.list is necessary, nor the `list` here. IIRC `zfunc("select_categories", derivation, value=I(selections))` is what you want, and then you could drop the if is.list check.